src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -333,7 +334,13 @@ public bool GetBoolean()
             return
                 type == JsonTokenType.True ? true :
                 type == JsonTokenType.False ? false :
-                throw ThrowHelper.GetJsonElementWrongTypeException(nameof(Boolean), type);
+                ThrowJsonElementWrongTypeException(type);
+
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            static bool ThrowJsonElementWrongTypeException(JsonTokenType actualType)
+            {
+                throw ThrowHelper.GetJsonElementWrongTypeException(nameof(Boolean), actualType.ToValueKind());
+            }
```

Comment:
Should this method be moved to ThrowHelper?

Reply:
It's leaking the return type of the expression, so I though it preferable to keep as a local. Moving it to ThrowHelper would need changing the return type to `void`, which forces an awkward refactoring in this method. No strong opinion either way, I can change it.